### PR TITLE
Legacy compatibility

### PR DESCRIPTION
**Pull request by ettomatic**
Fix issue with different signatures between Ruby 2.1 and 2.4.

For non-String scalar values, the object class name gets injected in the
signature data.

Ruby 2.4 unified Fixnum and Bignum into Integer so a signature
containing numbers in 2.1 will be different from one generated in 2.4.

Ruby 2.1
```ruby
2.class.ancestors
[Fixnum, Integer, Numeric, Comparable, Object, PP::ObjectMixin, Kernel, BasicObject]
```

Ruby 2.4.2

```ruby
2.class.ancestors
[Integer, Numeric, Comparable, Object, PP::ObjectMixin, Kernel, BasicObject]
```

Floats are not showing the same issues, as both 2.1 and 2.4 are returning:

```ruby
6.6.class.ancestors
[Float, Numeric, Comparable, Object, Kernel, BasicObject]
```

If you'll use Crimp to generate cache/database ids this can be a
problem.

This commit makes Crimp backwards compatible.

See also PR https://github.com/BBC-News/crimp/pull/8 for more context (thanks @coderkind and @Tarqwyn)

### CODE REVIEW

File: lib/crimp/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module Crimp
-  VERSION = '0.1.2'.freeze
+  VERSION = '1.0.0'.freeze
```
Comment (samfrench) [inline lib/crimp/version.rb]: Should we keep this change at 0.x and then have the new version at 1?

**Comment by samfrench** [comment]
Why do we require the version in the crimp.rb file when it doesn't use it?
> https://github.com/BBC-News/crimp/blob/master/lib/crimp.rb#L1

**Comment by ettomatic** [comment]
@samfrench  https://github.com/BBC-News/crimp/blob/dbc3db837c1e6d019fd779cc8e3447de2dedf532/crimp.gemspec#L8

File: lib/crimp/version.rb (hunk 1 above)
Comment (ettomatic) [reply lib/crimp/version.rb]: Pushed to 1.0.0 with the idea to set the new version as 2.0.0 but probably you're right, the current version is really not a 1.x thing!

File: spec/crimp_spec.rb [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
+  let(:hash_unordered) { { d: 'd', a: { c: 'c', b: 'b' } } }
+  let(:array) { [1, 2, 3, [4, [5, 6]]] }
+  let(:array_unordered) { [3, 2, 1, [[5, 6], 4]] }
 
   describe '.signature' do
     context 'given a Hash' do
-      specify { expect(subject.signature(hash)).to be_a String }
-
       it 'returns MD5 hash of stringified Hash' do
-        expect(subject.signature(hash)).to eq(Digest::MD5.hexdigest(subject.stringify(hash)))
```
Comment (ettomatic) [inline spec/crimp_spec.rb]: This is not testing anything 😢

Comment (junior451) [reply spec/crimp_spec.rb]: :(

**Comment by woodyblah** [comment]
@junior451 The public `signature` method doesn't need to store any state to work, so there's no need for an instance to be created. A static/class method can't call instance methods though, so all the private methods that `signature` calls will also need to be class methods in order for it to work (they also don't need state so this is fine)

File: lib/crimp.rb [hunk 3]
```diff
@@ -46,4 +48,19 @@ def self.parse_hash(hash)
...
+  # 2.class => Fixnum
+  # Ruby > 2.4
+  # 2.class => Integer
+  #
+  # Say you have a huge number of stored keys and you migrate your app from 2.1 to >= 2.4
+  # this would cause a change of the signature for a subset of the keys which would be hard
+  # to debug especially for nested data structures.
+  #
+  def self.integer_to_string(obj)
+    "#{obj}Fixnum"
```
Comment (coderkind) [inline lib/crimp.rb]: Probably an edge case, but would previous invocations _always_ be dealing with `Fixnum`'s (as opposed to `Bignum`'s as well)?

Comment (ettomatic) [reply lib/crimp.rb]: yes you're right... I'll add another _hack_  to handle this 😞 .

File: spec/crimp_spec.rb [hunk 4]
```diff
@@ -1,17 +1,16 @@
...
 describe Crimp do
-  let (:hash) { { a: { b: 'b', c: 'c' }, d: 'd' } }
-  let (:hash_unordered) { { d: 'd', a: { c: 'c', b: 'b' } } }
-  let (:array) { [1, 2, 3, [4, [5, 6]]] }
-  let (:array_unordered) { [3, 2, 1, [[5, 6], 4]] }
+  let(:hash) { { a: { b: 'b', c: 'c' }, d: 'd' } }
+  let(:hash_with_numbers) { { a: { b: 1, c: 3.14 }, d: 'd' } }
+  let(:hash_unordered) { { d: 'd', a: { c: 'c', b: 'b' } } }
+  let(:array) { [1, 2, 3, [4, [5, 6]]] }
+  let(:array_unordered) { [3, 2, 1, [[5, 6], 4]] }
```
[the remark this answers is not in the document]
Comment (ettomatic) [reply spec/crimp_spec.rb]: yes test coverage is weak, the new version of Crimp i'm working on has better tests tho.

**Pull request merged**